diff --git a/meson.build b/meson.build
new file mode 100644
index 0000000..e737d9a
--- /dev/null
+++ b/meson.build
@@ -0,0 +1,13 @@
+project('gnu_getopt', 'c', default_options : ['warning_level=0'])
+
+gnu_getopt_library = static_library(
+        'gnu_getopt',
+        files('src/getopt.c', 'src/getopt1.c'),
+        include_directories: include_directories('src/'),
+        install: false
+)
+
+gnu_getopt_dependency = declare_dependency(
+       link_with: gnu_getopt_library,
+       include_directories: include_directories('src/')
+)
+
